File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionUpper.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;

public class TransformerFunctionUpper extends TransformerFunction {
    public TransformerFunctionUpper() {
        super();
    }
    @Override
    public CompletionStage<Object> apply(FunctionContext context) {
        var result = context.getString(null);
        return result == null ? null : result.toUpperCase();
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionUpper.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";

class TransformerFunctionUpper extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    const result = await context.getString(null);
    return result == null ? null : result.toUpperCase();
  }
}

export default TransformerFunctionUpper;
